dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/persistence/file/FileExtTest.kt

```diff
@@ -451,6 +451,88 @@ internal class FileExtTest {
         assertThat(readText).isNull()
     }
 
+    @Test
+    fun `M return result W readBytesSafe()`(
+        @StringForgery result: String
+    ) {
+        // Given
+        fakeFile.writeText(result)
+
+        // When
+        val readBytes = fakeFile.readBytesSafe()
+
+        // Then
+        assertThat(readBytes).isEqualTo(result.toByteArray())
+    }
+
+    @Test
+    fun `M return result W readBytesSafe() {custom charset}`(
+        @StringForgery result: String,
+        @Forgery charset: Charset
+    ) {
+        // Given
+        fakeFile.writeText(result, charset)
+
+        // When
+        val readBytes = fakeFile.readBytesSafe()
+
+        // Then
+        assertThat(readBytes).isEqualTo(result.toByteArray(charset))
+    }
+
+    @Test
+    fun `M return null W readBytesSafe() {file doesn't exist}`() {
+        // Given
+        whenever(mockFile.exists()) doReturn false
+
+        // When
+        val readBytes = mockFile.readBytesSafe()
+
+        // Then
+        assertThat(readBytes).isNull()
+    }
+
+    @Test
+    fun `M return null W readBytesSafe() {file can't be read}`() {
+        // Given
+        whenever(mockFile.exists()) doReturn true
+        whenever(mockFile.canRead()) doReturn false
+
+        // When
+        val readBytes = mockFile.readBytesSafe()
+
+        // Then
+        assertThat(readBytes).isNull()
+    }
+
+    @Test
+    fun `M catch exception W readBytesSafe() {SecurityException}`(
+        @StringForgery message: String
+    ) {
+        // Given
+        whenever(mockFile.name) doThrow SecurityException(message)
+
+        // When
+        val readBytes = mockFile.readBytesSafe()
+
+        // Then
+        assertThat(readBytes).isNull()
+    }
+
+    @Test
+    fun `M catch exception W readBytesSafe() {FileNotFoundException}`() {
```

Comment:
You are missing the exception assertion here ?

Reply:
Method doesn't throw it, because it is swallowed internally by the `safeCall` wrapper and logged using `sdkLogger`. If we want to assert logger call, I would suggest doing it in another commit or PR, because then all similar methods in this file should have that. WDYT?

Reply:
Let's wait for the RUM telemetry topic to go around all the place were we actually want it and add the missing tests then.